rust/lance-index/src/vector/graph.rs

```diff
@@ -172,13 +172,11 @@ pub trait Graph {
 pub(super) fn beam_search(
     graph: &dyn Graph,
     start: &[OrderedNode],
-    query: &[f32],
     k: usize,
-    dist_calc: Option<Arc<dyn DistCalculator>>,
+    dist_calc: Arc<dyn DistCalculator>,
```

Comment:
if it is not optional, can we just use `&dyn DistCalculatr`?

Reply:
since `DistCalculator` is called in a tight loop. Maybe its worth making the calls static?

Reply:
@chebbyChefNEQ do you mean no `dyn`?

Reply:
I think he means that to use generic to avoid runtime dispatch.

```
pub fn beam_search(
   graph: &dyn Graph,
   ..
   dist_calc: impl DistCalcultor,
   bitset: ..
)
```